system_tests/nox.py

```diff
@@ -24,102 +24,124 @@
+    # if the CLOUD_SDK_ROOT is cached, it will need to be periodically cleared.
+    if CLOUD_SDK_INSTALL_DIR.exists():
         return
 
-    tar_file = 'google-cloud-sdk.tar.gz'
-    tar_path = CLOUD_SDK_ROOT.join(tar_file)
+    tar_path = CLOUD_SDK_ROOT.join(CLOUD_SDK_DIST_FILENAME)
 
     # Download the release.
-    prerun(
-        'wget', 'https://dl.google.com/dl/cloudsdk/release/{}'.format(
-            tar_file),
-        '-O', str(tar_path))
+    session.run(
+        'wget', CLOUD_SDK_DOWNLOAD_URL, '-O', str(tar_path), silent=True)
 
     # Extract the release.
-    prerun('tar', 'xzf', str(tar_path), '-C', str(CLOUD_SDK_ROOT))
-    tar_path.remove()
+    session.run(
+        'tar', 'xzf', str(tar_path), '-C', str(CLOUD_SDK_ROOT))
+    session.run(tar_path.remove)
 
     # Run the install script.
-    prerun(
-        str(CLOUD_SDK_ROOT.join('google-cloud-sdk', 'install.sh')),
+    session.run(
+        str(CLOUD_SDK_INSTALL_DIR.join('install.sh')),
         '--usage-reporting', 'false',
         '--path-update', 'false',
         '--command-completion', 'false',
-        env={CLOUD_SDK_CONFIG_ENV: str(CLOUD_SDK_ROOT)})
+        silent=True)
 
-    return CLOUD_SDK_ROOT
 
-
-def gcloud(*args, **kwargs):
-    """Calls the Cloud SDK CLI."""
-    prog = str(CLOUD_SDK_ROOT.join('google-cloud-sdk', 'bin', 'gcloud'))
-    env = {CLOUD_SDK_CONFIG_ENV: str(CLOUD_SDK_ROOT)}
-    return prerun(prog, *args, env=env, **kwargs)
-
-
-def configure_cloud_sdk(application_default_credentials, project=False):
-    """Configures the Cloud SDK with the given application default
+def configure_cloud_sdk(
+        session, application_default_credentials, project=False):
+    """Installs and configures the Cloud SDK with the given application default
     credentials.
 
     If project is True, then a project will be set in the active config.
     If it is false, this will ensure no project is set.
     """
+    install_cloud_sdk(session)
 
     if project:
-        gcloud('config', 'set', 'project', 'example-project')
+        session.run(GCLOUD, 'config', 'set', 'project', 'example-project')
     else:
-        gcloud('config', 'unset', 'project')
+        session.run(GCLOUD, 'config', 'unset', 'project')
 
     # Copy the credentials file to the config root. This is needed because
     # unfortunately gcloud doesn't provide a clean way to tell it to use
     # a particular set of credentials. However, this does verify that gcloud
     # also considers the credentials valid by calling application-default
     # print-access-token
-    dest = CLOUD_SDK_ROOT.join('application_default_credentials.json')
-    if dest.exists():
-        dest.remove()
-    py.path.local(application_default_credentials).copy(dest)
+    def copy_credentials():
+        dest = CLOUD_SDK_ROOT.join('application_default_credentials.json')
+        if dest.exists():
+            dest.remove()
+        py.path.local(application_default_credentials).copy(dest)
```

Comment:
:-1: on the closure. Does `session.run` not allow you to pass args to a function? If not, why not define `copy_credentials(app_creds)` as a global and then use `functools.partial(copy_credentials, app_creds)` to make your callable with no args.